Gems/ROS2/Code/Source/Manipulator/JointPublisherComponent.cpp

```diff
@@ -0,0 +1,118 @@
+#include <ROS2/Manipulator/JointPublisherComponent.h>
+#include <AzCore/Serialization/EditContext.h>
+#include "ROS2/ROS2Bus.h"
+#include <PhysX/Joint/PhysXJointRequestsBus.h>
+#include <Source/HingeJointComponent.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+
+namespace ROS2
+{
+    void JointPublisherComponent::Activate()
+    {
+        AZ::TickBus::Handler::BusConnect();
+        auto ros2Node = ROS2::ROS2Interface::Get()->GetNode();
+        AZStd::string jointStatesTopic = "joint_states";
+        m_jointstatePublisher = ros2Node->create_publisher<sensor_msgs::msg::JointState>(jointStatesTopic.data(), rclcpp::QoS(1));        // TODO: add QoS instead of "1"
+    }
+
+    void JointPublisherComponent::Deactivate()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void JointPublisherComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("JointPublisher"));
+    }
+
+    void JointPublisherComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("URDFMetadata"));
+    }
+
+    void JointPublisherComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<JointPublisherComponent, AZ::Component>()
+                ->Version(0);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<JointPublisherComponent>("JointPublisherComponent", "[Publish all the Hinge joint in the tree]")
```

Comment:
it should be addressed in the implementation of #171

Reply:
@adamdbrw Would you mind verifying that https://github.com/o3de/o3de-extras/issues/171 /  https://github.com/o3de/sig-simulation/issues/54 covers the open aspects of this discussion, so that we can close this before moving the PR forward?